File: Apps/app/src/core/domain/api/structures/Article.ts
import { IArticle } from './IArticle'
import { IAuthor } from './IAuthor'
import { ICategory } from './ICategory'
import { ITag } from './ITag'

const avgwpm = 40

export class Article implements IArticle
{
    author: IAuthor;
    authorId: string;
    category: ICategory;
    content: string;
    headerSrc: string;
    id: string;
    synopsis: string;
    tags: ITag[];
    title: string;
    slug: string;
    updatedAt: Date | null;
    writtenAt: Date;

    constructor(article: IArticle)
    {
        this.author = article.author
        this.authorId = article.authorId
        this.category = article.category
        this.content = article.content
        this.headerSrc = article.headerSrc
        this.id = article.id
        this.synopsis = article.synopsis
        this.tags = article.tags
        this.title = article.title
        this.slug = article.slug
        this.updatedAt = article.updatedAt
        this.writtenAt = article.writtenAt
    }

    get readTimeMinutes(): string
    {
        const words = this.content
            .split(' ')
            .length

        return Math.floor(words / avgwpm) + 'm'
    }
}

File: Services/Content/Domain/Content/Entities/Article.cs
namespace RunSynopsis.Domain.Content.Entities
{
    public class Article
    {
        public long Id { get; set; }

        public long AuthorId { get; set; }

        public long CategoryId { get; set; }

        public string HeaderSrc { get; set; }

        public string Title { get; set; }

        public string Slug { get; set; }

        public string Synopsis { get; set; }

        public string Content { get; set; }

        public DateTime WrittenAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public virtual Author Author { get; set; }

        public virtual Category Category { get; set; }

        public virtual ICollection<Tag> Tags { get; set; }
            = new List<Tag>();

        public Article()
        { }

        public Article(
            Author author,
            Category category,
            string headerSrc,
            string title,
            string slug,
            string synopsis,
            string content
        )
        {
            HeaderSrc = headerSrc;
            Title = title;
            Slug = slug;
            Synopsis = synopsis;
            Content = content;
            Author = author;
            AuthorId = author.Id;
            Category = category;
            CategoryId = category.Id;
        }
    }
}